### PR TITLE
Clean up legacy read interfaces

### LINKED ISSUE #436: Remove legacy read interfaces
Remove legacy read interfaces:

- INITPOST_GFS_NEMS
- INITPOST_NEMS_MPIIO
- INITPOST_NMM

### PR DESCRIPTION
**Pull request by WenMeng-NOAA**
Remove the following legacy read interfaces:

- INITPOST_GFS_NEMS
- INITPOST_NEMS_MPIIO
- INITPOST_NMM

### CODE REVIEW

**Comment by fossell** [comment]
@WenMeng-NOAA - thanks for adding serial netcdf GFS back.  We did support binarynemsmpiio (INITPOST_GFS_NEMS_MPIIO) for the original MRW release.  I'm not sure if MRW is planning to support that for next release, I would suspect no, so removing this would be ok. I can ask around to confirm unless you know?

**Comment by WenMeng-NOAA** [comment]
@fossell "INITPOST_GFS_NEMS_MPIIO" is a typo is my PR statement. It should be "INITPOST_NEMS_MPIIO".  Thanks for catching it.

**Comment by fossell** [comment]
@WenMeng-NOAA - Ok, yes I see that now in the file changes, I should have completed my review first before commenting.

**Review by fossell — APPROVED** [review]

File: sorc/ncep_post.fd/WRFPOST.f [hunk 1]
```diff
@@ -648,16 +616,10 @@ PROGRAM WRFPOST
           IF(MODELNAME == 'NCAR' .OR. MODELNAME == 'RAPR') THEN
             print*,'CALLING INITPOST TO PROCESS NCAR NETCDF OUTPUT'
             CALL INITPOST
-          ELSE IF(MODELNAME == 'NMM') THEN
```
Comment (HuiyaChuang-NOAA) [inline sorc/ncep_post.fd/WRFPOST.f:651]: @WenMeng-NOAA @fossell Is DTC still supporting WRF NMM users using NetCDF output?

Comment (fossell) [reply sorc/ncep_post.fd/WRFPOST.f:651]: No, DTC stopped support of WRF-NMM a few years ago. It's ok to remove this.

Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/WRFPOST.f:651]: Thanks for clarifying. I will remove it.

**Comment by WenMeng-NOAA** [comment]
The UPP regression tests were conducted on WCOSS-Dell, WCOSS-Cray and Hera. No changed results were found.

**Review by HuiyaChuang-NOAA — APPROVED** [review]

**Pull request merged**